core/src/macosX64Main/kotlin/Advertisement.kt

```diff
@@ -14,4 +21,39 @@ public actual data class Advertisement(
 
     public actual val name: String?
         get() = cbPeripheral.name
+
+    public actual val txPower: Int?
+        get() = (data[CBAdvertisementDataTxPowerLevelKey] as? NSNumber)?.intValue
+
+    actual val uuids: List<Uuid>
+        get() = (data[CBAdvertisementDataServiceUUIDsKey] as? List<CBUUID>)?.map { it.toUuid() } ?: emptyList()
+
+    public actual fun serviceData(uuid: Uuid): ByteArray? =
+        serviceDataAsNSData(uuid)?.toByteArray()
+
+    public fun serviceDataAsNSData(uuid: Uuid): NSData? =
+        (data[CBAdvertisementDataServiceDataKey] as? Map<CBUUID, NSData>)?.get(uuid.toCBUUID())
+
+    public actual fun manufacturerData(companyIdentifierCode: Int): ByteArray? =
+        manufacturerData?.takeIf { (it.code == companyIdentifierCode) }?.let { it.data }
+
+    public actual val manufacturerData: ManufacturerData?
+        get() = manufacturerDataAsNSData?.toByteArray()?.toManufacturerData()
+
+    public fun manufacturerDataAsNSData(companyIdentifierCode: Int): NSData? =
+        manufacturerData(companyIdentifierCode)?.toNSData()
+
+    public val manufacturerDataAsNSData: NSData?
+        get() = data[CBAdvertisementDataServiceDataKey] as? NSData
```

Comment:
Wrong key. This is why we were seeing `null` for `code` and `data`.

```suggestion
        get() = data[CBAdvertisementDataManufacturerDataKey] as? NSData
```

Thankfully the tests led me to investigate what data we were actually getting back, and voila! 😄

Reply:
OMG thanks for spotting this. I remember making sure to use the correct key originally and must have accidentally messed it up at some point 😭 Apple's bag-of-things API for advertisements is so vexing.

Reply:
I got it wrong in the tests also, at least my stupidity is consistent! 😑